gointelowl/tags.go

```diff
@@ -0,0 +1,146 @@
+	requestUrl := fmt.Sprintf("%s/api/tags", tagService.client.options.Url)
+	fmt.Println(requestUrl)
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	var tagList []Tag
+	json.Unmarshal(successResp.Data, &tagList)
+	return &tagList, nil
+}
+
+// * Getting a tag through its ID
+func (tagService *TagService) Get(ctx context.Context, tagId uint64) (*Tag, error) {
+	if err := checkTagID(tagId); err != nil {
+		return nil, err
+	}
+	requestUrl := fmt.Sprintf("%s/api/tags/%d", tagService.client.options.Url, tagId)
+	fmt.Println(requestUrl)
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+	var tagResponse Tag
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	json.Unmarshal(successResp.Data, &tagResponse)
+	return &tagResponse, nil
+}
+
+// //* Creating a Tag
+func (tagService *TagService) Create(ctx context.Context, tagParams *TagParams) (*Tag, error) {
+	requestUrl := fmt.Sprintf("%s/api/tags", tagService.client.options.Url)
+	fmt.Println("Url: " + requestUrl)
+
+	tagJson, err := json.Marshal(tagParams)
+	fmt.Println(string(tagJson))
+	if err != nil {
+		return nil, err
+	}
+	request, err := http.NewRequest("POST", requestUrl, bytes.NewBuffer(tagJson))
+	if err != nil {
+		return nil, err
+	}
+	var createdTag Tag
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	json.Unmarshal(successResp.Data, &createdTag)
+	return &createdTag, nil
+}
+
+//* Updating a tag
+func (tagService *TagService) Update(ctx context.Context, tagId uint64, tagParams *TagParams) (*Tag, error) {
+	requestUrl := fmt.Sprintf("%s/api/tags/%d", tagService.client.options.Url, tagId)
+	// printing the request
+	fmt.Println("Url: " + requestUrl)
+
+	// Getting the relevant JSON data
+	tagJson, err := json.Marshal(tagParams)
+	if err != nil {
+		return nil, err
+	}
+	request, err := http.NewRequest("PUT", requestUrl, bytes.NewBuffer(tagJson))
+	if err != nil {
+		return nil, err
+	}
+	var updatedTag Tag
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	json.Unmarshal(successResp.Data, &updatedTag)
+	return &updatedTag, nil
+}
+
+//* Deleting a tag
+func (tagService *TagService) Delete(ctx context.Context, tagId uint64) (bool, error) {
+	if err := checkTagID(tagId); err != nil {
+		return false, err
+	}
+	requestUrl := fmt.Sprintf("%s/api/tags/%d", tagService.client.options.Url, tagId)
+	// printing the request
+	fmt.Println("Url: " + requestUrl)
+	request, err := http.NewRequest("DELETE", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == 204 {
```

Comment:
We should change this to http.StatusNoContent

Reply:
Will do!